python/ray/air/_internal/accelerator_utils/device_manager.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import List
+import torch
+
+
+class TorchDeviceManager(ABC):
+    """This class contains the function needed for supporting
+    an acclerator family in Ray AI Library.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def get_accelerator_name() -> str:
+        """Gets the corresponding accelerator type, e.g. GPU, NPU."""
+
+    @staticmethod
+    @abstractmethod
+    def get_device_type() -> str:
+        """Gets the device type in deeplearning framwork,
+        e.g. cuda, hpu, npu in torch.
+        """
+
+    @staticmethod
+    @abstractmethod
+    def get_device_module():
```

Comment:
The interface doesn't makes sense to me. We cannot assume all accelerator framework to have the `*.device()` api. The only place we use `get_device_module` is to empty cache during `_shutdown_torch`.  

Having `get_devices` API should be enough. For the cleanup, we can probably propose a `DeviceManager.cleanup` api for each accelerator type instead.

Reply:
Agreed, just as mentioned earlier, the concept of device_module has been removed, and now Ray Train interacts only with the `DeviceManager` object.